code/modules/mob/living/carbon/xenomorph/castes/queen/abilities_queen.dm

```diff
@@ -448,3 +448,85 @@
 	if (get_dist(owner, receiver) > 7)
 		// Out of screen transfer.
 		owner.balloon_alert(owner, "Transferred plasma")
+
+
+#define BULWARK_LOOP_TIME 1 SECONDS
+#define BULWARK_RADIUS 4
+#define BULWARK_ARMOR_MULTIPLIER 0.25
+
+/datum/action/xeno_action/bulwark
+	name = "Royal Bulwark"
+	action_icon_state = "bulwark"
+	desc = "Creates a field of defensive energy, filling chinks in the armor of nearby sisters, making them more resilient."
+	plasma_cost = 100
+	cooldown_timer = 20 SECONDS
+	keybinding_signals = list(
+		KEYBINDING_NORMAL = COMSIG_XENOABILITY_QUEEN_BULWARK,
+	)
+	/// assoc list xeno = armor_diff
+	var/list/armor_mod_keys = list()
+
+/datum/action/xeno_action/bulwark/action_activate()
+	var/list/turf/affected_turfs = RANGE_TURFS(BULWARK_RADIUS, owner)
+	add_cooldown()
+
+	for(var/turf/target AS in affected_turfs)
+		//yes I realize this adds and removes it every move but its simple
+		//also we use this and not aura because we want speedy updates on entering
+		RegisterSignal(target, COMSIG_ATOM_EXITED, PROC_REF(remove_buff))
+		RegisterSignal(target, COMSIG_ATOM_ENTERED, PROC_REF(apply_buff))
+		ADD_TRAIT(target, TRAIT_BULWARKED_TURF, XENO_TRAIT)
+		for(var/mob/living/carbon/xenomorph/xeno in target)
```

Comment:
is it easier to just range check the hive list? That would also mean you're not picking up other hives, in the blue moon that's relevant.